src/core/color.js

```diff
@@ -162,6 +162,159 @@ Object.assign(pc, (function () {
+            };
+        }())
+    });
+
+    /**
+     * @field
+     * @static
+     * @readonly
+     * @name pc.Color.CYAN
+     * @type {pc.Color}
+     * @description A constant color set to cyan [0, 1, 1, 1].
+     */
+    Object.defineProperty(Color, 'CYAN', {
+        get: (function () {
+            var cyan = new Color(0, 1, 1, 1);
+            return function () {
+                return cyan;
+            };
+        }())
+    });
+
+     /**
+      * @field
+      * @static
+      * @readonly
+      * @name pc.Color.GRAY
+      * @type {pc.Color}
+      * @description A constant color set to gray [0.5, 0.5, 0.5, 1].
+      */
+    Object.defineProperty(Color, 'GRAY', {
+        get: (function () {
+            var gray = new Color(0.5, 0.5, 0.5, 1);
+            return function () {
+                return gray;
+            };
+        }())
+    });
+
+     /**
+      * @field
+      * @static
+      * @readonly
+      * @name pc.Color.GREEN
+      * @type {pc.Color}
+      * @description A constant color set to green [0, 1, 0, 1].
+      */
+    Object.defineProperty(Color, 'GREEN', {
+        get: (function () {
+            var green = new Color(0, 1, 0, 1);
+            return function () {
+                return green;
+            };
+        }())
+    });
+
+     /**
+      * @field
+      * @static
+      * @readonly
+      * @name pc.Color.MAGENTA
+      * @type {pc.Color}
+      * @description A constant color set to magenta [1, 0, 1, 1].
+      */
+    Object.defineProperty(Color, 'MAGENTA', {
+        get: (function () {
+            var magenta = new Color(1, 0, 1, 1);
+            return function () {
+                return magenta;
+            };
+        }())
+    });
+
+     /**
+      * @field
+      * @static
+      * @readonly
+      * @name pc.Color.RED
+      * @type {pc.Color}
+      * @description A constant color set to red [1, 0, 0, 1].
+      */
+    Object.defineProperty(Color, 'RED', {
+        get: (function () {
+            var red = new Color(1, 0, 0, 1);
+            return function () {
+                return red;
+            };
+        }())
+    });
+
+     /**
+      * @field
+      * @static
+      * @readonly
+      * @name pc.Color.WHITE
+      * @type {pc.Color}
+      * @description A constant color set to white [1, 1, 1, 1].
+      */
+    Object.defineProperty(Color, 'WHITE', {
+        get: (function () {
+            var white = new Color(1, 1, 1, 1);
+            return function () {
+                return white;
+            };
+        }())
+    });
+
+     /**
+      * @field
+      * @static
+      * @readonly
+      * @name pc.Color.YELLOW
+      * @type {pc.Color}
+      * @description A constant color set to yellow [1, 1, 0, 1].
+      */
+    Object.defineProperty(Color, 'YELLOW', {
+        get: (function () {
+            var yellow = new Color(1, 1, 0, 1);
+            return function () {
+                return yellow;
+            };
+        }())
+    });
+
```

Comment:
You could probably condense this considerably by creating a `defineColor` function and calling it with name + value parameters.

Reply:
done

Reply:
Definitely more succinct which is great. But the updated approach loses the read-only nature of these properties. I was thinking something like:

```
var _defineColor = new function (name, r, g, b) {
    Object.defineProperty(Color, name, {
        get: (function () {
            var color = new Color(r, g, b);
            return function () {
                return color;
            };
        }())
    });
};

_defineColor('WHITE', 1, 1, 1);
_defineColor('BLACK', 0, 0, 0);
```

What do you think?

Reply:
LOL, actually, `writeable` seems to default to false. So actually, what you have might well be fine actually! Can you confirm?

Reply:
yes writeable is false based on documentation. I also tried writing to it .. and the value does not change, so all seems good.
I'd say we change Vec3.ONE, ZERO and few others to this as well?

Reply:
Yes, I was thinking the same thing. This looks like a better style on which to standardize!